Add advanced concepts to about page (#5922)
opensearch-project/documentation-website: about.md

```

You can change most OpenSearch settings using the REST API, modify indexes, check the health of the cluster, get statistics---almost everything.

## Advanced concepts

The following section describes more advanced OpenSearch concepts.

### Translog

> Any index changes, such as indexing or deleting documents, are written to disk during a Lucene commit. However, Lucene commits are expensive operations, so they cannot be performed after every change to the index. Instead, each shard records every indexing operation in a transaction log called translog. When a document is indexed, it is added to the memory buffer and recorded in the translog. Frequent refresh operations write the documents in the memory buffer to a segment and then clear the memory buffer. Periodically, a flush performs a Lucene commit, which includes writing the segments to disk using `fsync`, purging the old translog, and starting a new translog. Thus, a translog contains all operations that have not yet been flushed.

Reviewer: > When a document is indexed, it is added to the memory buffer and recorded in the translog.

Maybe it's kind of implied, but nothing explicitly mentions that the translog is durable but the "in memory buffer" is not. A naive reading of this doesn't explain why the document is recorded in both places. (The answer is that the translog is durably written to disk and can survive a process or host restart, but any data in the in memory buffer will be lost.)

### Refresh

Periodically, OpenSearch performs a refresh operation, which writes the documents from the in-memory Lucene index to files. These files are not durable because they reside in the file cache. A refresh makes documents available for search.

### Flush

A flush operation persists the files from the file cache to disk, ensuring durability. Flushing the files ensures that the data stored only in the translog is recorded in the Lucene index. OpenSearch performs a flush as needed using `fsync`. After `fsync`, segment files become immutable because they have been written to disk. 

### Merge

> In OpenSearch, a shard is a Lucene index, which consists of segments (or segment files). Segments store the indexed data. Periodically, smaller segments are merged into larger ones and the larger segments become immutable. This is done in order to prevent having many small segments on disk. The merge policy specifies how often merges are performed. Merging reduces the overall number of segments on each shard and frees up disk space.

Reviewer: > the larger segments become immutable

I think this is confusing. The smaller ones are also immutable, which is why they are merged and replaced by new segments as opposed to being updated in place. Are you trying to say that eventually segments reach a maximum size per the merge policy and are no longer merged into larger segments?

> This is done in order to prevent having many small segments on disk

Probably worth explaining why many small segments is undesirable by appending something like "which reduces search performance" to this sentence.

```

You can change most OpenSearch settings using the REST API, modify indexes, check the health of the cluster, get statistics---almost everything.

## Advanced concepts

The following section describes more advanced OpenSearch concepts.

### Translog

> Any index changes, such as indexing or deleting documents, are written to disk during a Lucene commit. However, Lucene commits are expensive operations, so they cannot be performed after every change to the index. Instead, each shard records every indexing operation in a transaction log called _translog_. When a document is indexed, it is added to the memory buffer and recorded in the translog. After a process or host restart, any data in the in-memory buffer is lost. Recording the document in the translog ensures durability because the translog is written to disk.

Reviewer 2: ```suggestion
Any index changes, such as document indexing or deletion, are written to disk during a Lucene commit. However, Lucene commits are expensive operations, so they cannot be performed after every change to the index. Instead, each shard records every indexing operation in a transaction log called _translog_. When a document is indexed, it is added to the memory buffer and recorded in the translog. After a process or host restart, any data in the in-memory buffer is lost. Recording the document in the translog ensures durability because the translog is written to disk.
```

Frequent refresh operations write the documents in the memory buffer to a segment and then clear the memory buffer. Periodically, a [flush](#flush) performs a Lucene commit, which includes writing the segments to disk using `fsync`, purging the old translog, and starting a new translog. Thus, a translog contains all operations that have not yet been flushed.

### Refresh

> Periodically, OpenSearch performs a _refresh_ operation, which writes the documents from the in-memory Lucene index to files. These files are not durable because they reside in the file cache. A refresh makes documents available for search.

Reviewer: ```suggestion
Periodically, OpenSearch performs a _refresh_ operation, which writes the documents from the in-memory Lucene index to files. These files are not guaranteed to be durable because an `fsync` is not performed. A refresh makes documents available for search.
```

### Flush

> A _flush_ operation persists the files to disk using `fsync`, ensuring durability. Flushing ensures that the data stored only in the translog is recorded in the Lucene index. OpenSearch performs a flush as needed to ensure the translog does not grow too large.

Reviewer 2: ```suggestion
A _flush_ operation persists the files to disk using `fsync`, ensuring durability. Flushing ensures that the data stored only in the translog is recorded in the Lucene index. OpenSearch performs a flush as needed to ensure that the translog does not grow too large.
```

> A _flush_ operation persists the files from the file cache to disk, ensuring durability. Flushing the files ensures that the data stored only in the translog is recorded in the Lucene index. OpenSearch performs a flush as needed using `fsync`. After `fsync`, segment files become immutable because they have been written to disk. 

Reviewer: ```suggestion
A _flush_ operation persists the files to disk using `fsync`, ensuring durability. Flushing ensures that the data stored only in the translog is recorded in the Lucene index. OpenSearch performs a flush as needed to ensure the translog does not grow too large.
```
